deploy/awstools/awstools.py

```diff
@@ -344,19 +344,32 @@ def auto_create_bucket(userbucketname):
             rootLogger.critical(repr(exc))
             assert False
 
-def subscribe_to_firesim_topic(email):
-    """ Subscribe a user to their FireSim SNS topic for notifications. """
+def get_snsname_arn():
+    """ If the Topic doesn't exist create it, send catch exceptions while creating. Or if it exists get arn """
     client = boto3.client('sns')
 
     aws_resource_names_dict = aws_resource_names()
     snsname = aws_resource_names_dict['snsname']
 
-    # this will either create the topic, if it doesn't exist, or just get the arn
-    response = client.create_topic(
-        Name=snsname
-    )
-    arn = response['TopicArn']
+    response = None
+    try: # this will either create the topic, if it doesn't exist, or just get the arn
+        response = client.create_topic(
+            Name=snsname
+        )
+    except client.exceptions.ClientError as err:
+        if 'AuthorizationError' in repr(err): 
+            rootLogger.critical("You don't have permissions to perform \"Topic Creation \". Required to send you email notifications. Please contact your IT administrator")
+            rootLogger.critical(err)
+        else:
+            rootLogger.critical("Unknown exception is encountered while trying to perform \"Topic Creation\"")
+        assert False
```

Comment:
```suggestion
            rootLogger.critical("You don't have permissions to perform \"Topic Creation \". Required to send you email notifications. Please contact your IT administrator")
        else:
            rootLogger.critical("Unknown exception is encountered while trying to perform \"Topic Creation\"")
        rootLogger.critical(err)
        assert False
```

Ugh, github's rendering of this patch is really ugly, should just be moving a line down with an indent change.  Anyway, move the `rootLogger.critical(err)` to just before the `assert False` so that it is logged either way.

Reply:
Done